modules/vectra.py

```diff
@@ -647,20 +651,11 @@ def advanced_search(self, stype=None, page_size=50, query=None):
         """
         if stype not in ["hosts", "detections"]:
             raise ValueError("Supported values for stype are hosts or detections")
-        return requests.get('{url}/search/{stype}/?page_size={ps}&query_string={query}'.format(url=self.url, stype=stype,
-                                                ps=page_size, query=query),
-                            headers=self.headers, verify=self.verify)
-
-    @request_error_handler
-    def custom_endpoint(self, path=None, **kwargs):
-        if not str(path).startswith('/'):
-            path = '/' + str(path)
 
-        params = {}
-        for k, v in kwargs.items():
-            params[k] = v
-
-        if self.version == 2:
-            return requests.get(self.url + path, headers=self.headers, params=params, verify=self.verify)
-        else:
-            return requests.get(self.url + path, auth=self.auth, params=params, verify=self.verify)
+        url = '{url}/search/{stype}/?page_size={ps}&query_string={query}'.format(url=self.url, stype=stype,
+                                                ps=page_size, query=query)
+        resp = self._get_request(url=url)
+        yield resp.json()['results']
+        while resp.json()['next']:
+            resp = self._get_request(url=resp.json()['next'])
+            yield resp.json()['results']
```

Comment:
This changes the object returned to be the results list instead of the (request) response object.  This may break existing customer code using this library and using the advanced_search method.

Reply:
I agree on the principle; however, the original code is bugged. For instance, in the get_rule(name=xxx) if the customer has more rules than returned on one page this fails to find a rule matching the name. 

IMO we could indeed find another fix that doesn't break backwards compatibility, but the current version is

Reply:
We should be able to fix it without breaking backwards compatibility